src/sys/resource.rs

```diff
@@ -295,7 +295,10 @@ impl Usage {
 
     /// The resident set size at its peak, in kilobytes.
     pub fn max_rss(&self) -> c_long {
-        self.0.ru_maxrss
+        #[cfg(not(apple_targets))]
+        return self.0.ru_maxrss;
+        #[cfg(apple_targets)]
+        return self.0.ru_maxrss >> 10;
```

Comment:
What about updating the doc on `apple_targets` rather than tweaking the code so that we can be more resilient to this kind of [changes](https://github.com/nix-rust/nix/issues/2331#issuecomment-2001858705):

```rs
    /// The resident set size at its peak,
    #[cfg_attr(apple_targets, doc = " in bytes.")]
    #[cfg_attr(not(apple_targets), doc = "in kilobytes.")]
    pub fn max_rss(&self) -> c_long {
        self.0.ru_maxrss
```